src/figuresClasses.ts

```diff
@@ -1,11 +1,77 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sides: number[] = [this.a, this.b, this.c].sort((first, second) => second - first);
+
+    if (sides[0] >= sides[1] + sides[2]) {
+      throw new Error('The sides do not form a valid triangle');
+    }
+
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('Put number bigger than 0');
```

Comment:
The error message 'Put number bigger than 0' is not very descriptive. Consider rephrasing it to something like 'Side lengths must be greater than 0'.